File: cpp/orden-expov2.cpp
#include <iostream>
#include <string>
#include "./module/linkedlist.h"
using namespace std;

LinkedList<string> split(string text,string sep=" ")
{
    LinkedList<string> llist;
    int start = 0;
    int end = text.find(sep);
    while (end != -1){
        llist.add(text.substr(start, end));
        start = end + sep.size();
        end = text.find(sep, start);
    };
    llist.add(text.substr(start, end));
    // llist.add(text.substr(start, end - start));
    return llist;
}

void ordenExpo(LinkedList<string>& llist1,LinkedList<string>& llist2)
{
    int count1 = 0;
    int count2 = llist1.getSize() - 1;
    bool flag = false;
    while (count1 <= count2)
    {
        if (flag)
        {
            llist2.add(llist1.get(count2));
            count2--;
            flag = false;
            continue;
        }
        llist2.add(llist1.get(count1));
        count1++;
        flag = true;
    }
}

int main(){
    string name;
    LinkedList<string> expo;
    getline(cin, name);
    LinkedList<string> person = split(name);
    ordenExpo(person, expo);
    cout << expo.toString();    
    return 0;
}

File: python/orden-expov2.py
from module.doublylinked import Doubly_linked_list
from module.node import Node

def split(text:str, sep:str = ' ') -> Doubly_linked_list:
    llist:Doubly_linked_list = Doubly_linked_list()
    start = 0
    end = text.find(sep)
    while end != -1:
        llist.add(text[start:end])
        start = end + len(sep)
        end = text.find(sep, start)
    llist.add(text[start:len(text)])
    return llist
    
    
def orden_expo(list1: Doubly_linked_list) -> Doubly_linked_list:
    lexpo:Doubly_linked_list = Doubly_linked_list()
    first: Node = list1.head
    end: Node = list1.tail
    change:bool = False
    while first != end:
        if(change):
            lexpo.add(end.data)
            end = end.prev
            change = False
            continue
        lexpo.add(first.data)
        first = first.next
        change = True
    lexpo.add(end.data)
    return lexpo

def main() -> None:
    nickname:str = input()
    lperson: Doubly_linked_list = split(nickname)
    lexpo: Doubly_linked_list = orden_expo(lperson)
    print(f"{lexpo}",end="")
    c: Doubly_linked_list = lexpo.copy()
    c.add("Z")
    print(f"\n{c}")
    print(f"{lexpo}",end="")

if __name__ == "__main__":
    main()